units/host_monitoring_idle_termination.go

```diff
@@ -263,12 +269,15 @@ func (i hostIdleInfo) getTerminationReason() string {
 		// busy.
 		return "host has an outdated AMI"
 	}
-	if i.timeSinceLastCommunication >= i.idleThreshold {
+	if i.timeSinceLastCommunication >= i.idleThreshold && !i.isRunningTearDownTaskGroup {
 		return fmt.Sprintf("host is idle or unreachable, communication time %s is over threshold time %s", i.timeSinceLastCommunication, i.idleThreshold)
 	}
-	if i.idleTime >= i.idleThreshold {
+	if i.idleTime > 0 && i.idleTime >= i.idleThreshold {
```

Comment:
Is this check meant for `idleThreshold`? Assuming `idleThreshold` > 0, wouldn't the 2nd if condition always encapsulate the first? Can `idleThreshold` be set to 0 and if so, why (maybe worth a comment here to explain this niche case)

Reply:
idleThreshold is set to zero when the task is running teardown, since the task does not send heartbeats during teardown, so that's why this check is necessary. It wouldn't be encapsulated by the 2nd condition because of the `!i.isRunningTearDownTaskGroup` check.